src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
+
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int RESIZE_MULTIPLIER = 2;
+
+    private int threshold;
+    private int size;
+    private int capacity;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+        this.capacity = DEFAULT_INITIAL_CAPACITY;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        putNode(calculateHash(key), key, value);
```

Comment:
The method should have as less arguments as possible
You can **calculateHash** inside **putNode**. No any reason to do it before **putNode**
```suggestion
        putNode(key, value);
```